dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ServiceAnnotationBeanPostProcessor.java

```diff
@@ -447,6 +455,12 @@ private AbstractBeanDefinition buildServiceBeanDefinition(Annotation serviceAnno
 
     }
 
+    private List convertMethodConfigs(Object methodsAnnotation) {
+        if (methodsAnnotation == null){
+            return Collections.EMPTY_LIST;
+        }
+        return MethodConfig.constructMethodConfig((Method[])methodsAnnotation);
+    }
 
```

Comment:
The code seems a little complexed, I think
```java
Optional.ofNullable(serviceAnnotationAttributes.get("methods")).ifPresent(methods -> builder.addPropertyValue("methods", MethodConfig.constructMethodConfig((Method[]) methods)));
```
one line of code is enough.

Just for reference

Reply:
@Leishunyu Do you think it's better to use this to replace the origin implementation ? I think it's simpler and cleaner ?

Reply:
I think the code is simple, but the method semantics are not clear enough.According to your way, I think most methods can be rewritten, but this is not necessary.What do you think？